pkg/controller.v1/common/job.go

```diff
@@ -421,8 +421,8 @@ func (jc *JobController) CleanupJob(runPolicy *apiv1.RunPolicy, jobStatus apiv1.
 	currentTime := time.Now()
 	metaObject, _ := job.(metav1.Object)
 	ttl := runPolicy.TTLSecondsAfterFinished
-	if ttl == nil {
-		return nil
+	if ttl == nil || commonutil.IsSuspended(jobStatus) { 
```

Comment:
```suggestion
	if ttl == nil || trainutil.IsJobSuspended(runPolicy) { 
```

Sorry for your confusion. I guess that we need to check the runPolicy instead of status since the suspension could be issued before the status is updated.

Reply:
Right, good point, I should have notice that.
Thanks!